src/data/markdown/docs/01 guides/05 Testing Guides/04 Large-scale k6 testing.md

```diff
@@ -0,0 +1,484 @@
+
+```
+WARN[0034] Request Failed     error="Get http://99.81.83.131/static/logo.svg?ip=6: dial tcp 99.81.83.131:80: socket: too many open files"
+```
+
+</div>
+
+
+Note: you should decide what level of errors is acceptable. At large scale some errors are always present.
+If you make 50M requests with 100 failures, this is generally a good result (0.00002% errors).
+
+# Benchmarking k6 on AWS hardware
+
+We have executed few large tests on different EC2 machines to see how much load k6 can generate. 
+Our general observation is that k6 scales proportionally to the hardware. 2x larger machine is able to generate 2x more traffic.
+The limit to this scalability is in the number of open connections. Single Linux machine can open up to `65 535` sockets per IP. 
+This means that maximum of 65k requests can be executed simultaneously on a single machine.
+The RPS limit depends on the response time of the SUT. If responses are delivered in 100ms, the RPS limit is 650 000. 
+
+## Real-life test of a website.
+
+Testing the theoretical limits is fun, but that's not the point of this benchmark. 
+The point of this benchmark is to give users indication of how much traffic k6 can generate when executing complicated, real-life tests.
+For this purpose we have written a rather heavy [real-life website test](https://github.com/loadimpact/k6-hardware-benchmark/blob/master/scripts/website.js) that uses almost all k6 features.
+
+Setup:
+- All tests were executed on AWS EC2 instances
+- The "discardResponseBodies" recommendation was NOT used. (results would be better with this setting).
+- Scripts used for testing are available in the `/scripts` directory. The results are reproducible
+- k6 v0.26.2 was used 
+- Note: the target system (test.k6.io) was running on a large cluster to boost performance. 
+- Note: the target system (test.k6.io) is a slow-ish PHP website, not optimized for performance - a static website would be much quicker.
+
+The "website.js" test file uses a wide range of k6 features to make the test emulate a real usage of k6. This is not a test rigged for performance - quite the opposite.
+This test uses plenty of custom metrics, checks, parametrization, batches, thresholds and groups. It's a heavy test that should represent well the "real life" use case.
+
+### Execution on AWS m5.large EC2 server
+
+The `m5.large` instance has 8GB of RAM and 2 CPU cores. 
+
+The following command was used to execute the test
+<div class="code-group">
+
+```bash
+k6 run scripts/website.es5.js \
+ -o cloud \
+ --vus=6000 \
+ --duration=10m \
+ --compatibility-mode=base \
+ --no-thresholds \
+ --no-summary
+```
+
+</div>
+
+Results
+- Maximum VUS reached: 6000
+- Memory used: 6.09 GB  (out of 8.0)
+- CPU load (avg): 1.49 (out of 2.0). 
+- Peak RPS: ~6000 (note, this test was not optimized for RPS).
+- 2x `sleep(5)` in each iteration.
+
+### Execution on AWS m5.4xlarge
+The `m5.4xlarge` instance has 64GB of RAM and 16 CPU cores.
+https://app.k6.io/runs/720179
```

Comment:
Is this going to be a public link? Will the other ones have public links as well ?

Reply:
Ah, missed it. I'll remove these links.